packages/date-picker/src/DatePicker/components/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from "react"
+import { Icon } from "@kaizen/component-library"
+import classnames from "classnames"
+import {
+  FieldGroup,
+  FieldMessage,
+  Input,
+  InputProps,
+  Label,
+} from "@kaizen/draft-form"
+
+import { format, parse } from "date-fns"
+import { Modifier } from "react-day-picker"
+import { DateFormat } from "../../DatePicker"
+import { isInvalidDate } from "../../../utils/isInvalidDate"
+import { isDisabledDate } from "../../../utils/isDisabledDate"
+import styles from "./DateInput.scss"
+
+type OmittedInputProps =
+  | "startIconAdornment"
+  | "endIconAdornment"
+  | "inputType"
+  | "inputValue"
+  | "reversed"
+  | "onBlur"
+
+export interface validationMessagesProps {
+  error?: React.ReactNode
+}
+
+export interface DateInputProps extends Omit<InputProps, OmittedInputProps> {
+  id: string
+  calendarId: string
+  buttonRef?: React.RefObject<HTMLButtonElement>
+  labelText: React.ReactNode
+  icon: React.SVGAttributes<SVGSymbolElement>
+  /**
+   * A description that provides context for the text field
+   */
+  description?: React.ReactNode
+  isOpen: boolean
+  onButtonClick: () => void
+  isReversed?: boolean
+  valueDate: Date | undefined
+  onBlur: (date: Date | undefined) => void
+  disabledDays?: Modifier | Modifier[]
+}
+
+const handleFormatChange = (
+  parsedDate: Date,
+  disabledDays: Modifier | Modifier[],
+  setValueString: React.Dispatch<React.SetStateAction<string>>
+): void => {
+  if (!isInvalidDate(parsedDate) && !isDisabledDate(parsedDate, disabledDays)) {
+    setValueString(format(parsedDate, DateFormat.Text))
+  }
+}
```

Comment:
```suggestion
const formatDateAsText = (
  date: Date,
  disabledDays: Modifier | Modifier[],
  onValidDate: (string) => void
): void => {
  if (!isInvalidDate(date) && !isDisabledDate(date, disabledDays)) {
    onValidDate(format(date, DateFormat.Text))
  }
}
```

Reply:
(string) => void had an error

Reply:
Haha oops! It should've been `(newFormattedDate: string) => void` 😄